cmd/common/helpers.go

```diff
@@ -49,3 +57,67 @@ func CreateResourceLock(cb *clients.Builder, componentNamespace, componentName s
 		},
 	}
 }
+
+// TrustedCAWatcher is an helper type to setup a file watcher on the trusted CA
+type TrustedCAWatcher struct {
+	watcher  *fsnotify.Watcher
+	original []byte
+}
+
+// NewTrustedCAWatcher creates a new instance of the trusted CA watcher
+func NewTrustedCAWatcher() (trustedCAWatcher *TrustedCAWatcher, retErr error) {
+	// Set up and start the file watcher.
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create watcher")
+	}
+	defer func() {
+		if retErr != nil {
+			watcher.Close()
+		}
+	}()
+	if err := watcher.Add(defaultTrustedCABundle); err != nil {
+		return nil, errors.Wrapf(err, "failed to add file %q to watcher", defaultTrustedCABundle)
+	}
+	orig, err := ioutil.ReadFile(defaultTrustedCABundle)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to read watched file %q", defaultTrustedCABundle)
+	}
+	trustedCAWatcher = &TrustedCAWatcher{}
+	trustedCAWatcher.watcher = watcher
+	trustedCAWatcher.original = orig
+	return
+}
+
+// Close closes the internal watcher
+func (trustedCAWatcher *TrustedCAWatcher) Close() {
+	trustedCAWatcher.watcher.Close()
+}
+
+// Run loops listening to file events. Most of the time you'd want this to be run in a goroutine
+func (trustedCAWatcher *TrustedCAWatcher) Run(stopCh chan struct{}) {
+	for {
+		select {
+		case event := <-trustedCAWatcher.watcher.Events:
+			if event.Op == fsnotify.Remove {
+				trustedCAWatcher.watcher.Remove(event.Name)
+				trustedCAWatcher.watcher.Add(defaultTrustedCABundle)
+			}
+			latest, err := ioutil.ReadFile(defaultTrustedCABundle)
+			if err != nil {
+				glog.Errorf("failed to read watched file %q: %v", defaultTrustedCABundle, err)
+				close(stopCh)
+				return
+			}
+			if !bytes.Equal(trustedCAWatcher.original, latest) {
+				glog.Infof("watched file %q changed, stopping operator", defaultTrustedCABundle)
+				close(stopCh)
+				return
+			}
+		case err := <-trustedCAWatcher.watcher.Errors:
```

Comment:
Just to verify, if the watcher errors we expect to continue watching?

Reply:
Yep, can be transient and if it’s not we’ll see that in logs